### ISSUE TITLE
sklearn wrapper XGBClassifier fit method failed when len(y_values) is 2

### ISSUE THREAD

**Issue by xm11092** [dmlc/xgboost#2366]
## Environment info
Operating System:
windows

Package used (python/R/jvm/C++):
python 

`xgboost` version used:
0.6

## Steps to reproduce

param_test = {
        'max_depth': [3, 4, 5, 6],
    }
scoring = 'accuracy'
estimator = XGBClassifier(
	learning_rate=0.1,
	n_estimators=1000,
	objective='multi:softmax')
 gsearch = GridSearchCV(estimator, param_grid=param_test, scoring=scoring, n_jobs=4, iid=False)
 gsearch.fit(train_X, fitY)

when len(y_values) is 5 then it's OK, and if len(y_values) is 2, its failed and error is "value 0for Parameter num_class should be greater equal to 1".  But use xgboost.train api, then the issue not happen. 
How could I resolve the issue when using XGBClassifier which is needed for GridSearchCV
